modules/apps/commerce/commerce-shipment-web/src/main/java/com/liferay/commerce/shipment/web/internal/display/context/CommerceShipmentDisplayContext.java

```diff
@@ -384,6 +428,37 @@ public CommerceAddress getShippingAddress() throws PortalException {
 			commerceShipment.getCommerceAddressId());
 	}
 
+	public boolean hasMultipleShippingMethods() throws PortalException {
+		long commerceShippingMethodId = 0;
+
+		List<CommerceShipmentItem> commerceShipmentItems =
+			_commerceShipmentItemService.getCommerceShipmentItems(
+				getCommerceShipmentId(), QueryUtil.ALL_POS, QueryUtil.ALL_POS,
+				null);
+
+		for (CommerceShipmentItem commerceShipmentItem :
+				commerceShipmentItems) {
+
+			CommerceOrderItem commerceOrderItem =
+				_commerceOrderItemService.getCommerceOrderItem(
+					commerceShipmentItem.getCommerceOrderItemId());
+
+			CommerceOrder commerceOrder = commerceOrderItem.getCommerceOrder();
+
+			if ((commerceShippingMethodId != 0) &&
+				(commerceShippingMethodId !=
+					commerceOrder.getCommerceShippingMethodId())) {
+
+				return true;
+			}
+
+			commerceShippingMethodId =
+				commerceOrder.getCommerceShippingMethodId();
+		}
+
+		return false;
+	}
+
```

Comment:
This has been hard to read, but I get the point. Would it make sense to use Stream.distinct and then check on the size?